orttraining/orttraining/core/optimizer/megatron_transformer.h

```diff
@@ -15,13 +15,17 @@ class MegatronTransformer : public GraphTransformer {
                       std::unordered_map<std::string, std::string>& updated_weight_names,
                       std::unordered_set<std::string>& weights_to_train,
                       std::unordered_map<std::string, training::TrainingSession::PartitionInfo>& weight_partition_info,
+                      training::TrainingSession::OptimizerState& initial_optimizer_states,
```

Comment:
this is just a mapping we used in readonly mode, right? make it a const ref?

Reply:
we are modifying the initial optimizer states passed: we partition the moments the same way as the weight is partitioned. Thus cannot be const